File: Go/08d.go
package main

import (
	"fmt"
	"os"
)

//lint:ignore U1000 //

/*
func main() {
	eightd()
}

//*/

func eightd() {
	var s, p string
	fmt.Scanf("%s", &s)
	fmt.Scanf("%s", &p)
	for i := 0; i < len(s); i++ {
		for j := 0; j < len(p); j++ {
			if s[(i+j)%len(s)] != p[j] {
				break
			} else if j == len(p)-1 {
				fmt.Println("Yes")
				os.Exit(0)
			}
		}
	}
	fmt.Println("No")
}

File: Python/08d.py
# standard
s=input()
p=input()
for i in range(len(s)):
    for j in range(len(p)):
        if s[(i+j)%len(s)] != p[j]:
            break
        elif j == len(p)-1:
            print("Yes")
            exit()
print("No")

# shorter
'''
s=input()*2
p=input()
if p in s: print("Yes")
else: print("No")
'''